src/main/java/edu/ucla/library/iiif/auth/delegate/CantaloupeAuthDelegate.java

```diff
@@ -117,59 +115,27 @@ public Object preAuthorize() {
             case OPEN:
                 return true;
             case TIERED:
-                final Optional<HauthToken> token = getToken(authorizationHeader);
-
-                if (token.isPresent() && token.get().isValidIP()) {
-                    // Full access
-                    return true;
-                } else if (Arrays.equals(configuredScaleConstraint, scaleConstraint)) {
-                    // Degraded image request for the size we allow access to
-                    // (Probably via an earlier HTTP 302 redirect)
-                    myInfoJsonShouldContainAuth = true;
-
-                    return true;
-                } else if (scaleConstraint[0] != scaleConstraint[1]) {
-                    // Degraded image request for a size we don't allow access to; return HTTP 403
-                    return false;
-                } else {
-                    // Full image request, but non-campus IP
-                    // (The long types make a difference here, apparently)
-                    return Map.of(STATUS_CODE, Long.valueOf(HTTP.FOUND), //
-                            "scale_numerator", (long) configuredScaleConstraint[0], //
-                            "scale_denominator", (long) configuredScaleConstraint[1]);
+                switch (getRequestType()) {
+                    case INFORMATION:
+                        return getTieredInfo(authorizationHeader);
+                    case IMAGE:
+                    default:
+                        return getTieredImage();
```

Comment:
I think this helps a lot, btw.

Reply:
That handles the complexity of that method but gets us in trouble with the god class... can't win for losing 😆 .